### ISSUE TITLE
with_dict cannot recognize dict object

### ISSUE THREAD

**Issue by dimakievua** [ansible/ansible#20387]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
ansible Loops module

##### ANSIBLE VERSION
ansible 2.0.0.2

##### CONFIGURATION
No changes were made in ansible.cfg

##### OS / ENVIRONMENT
Ubuntu 5.4.0-6ubuntu1~16.04.4

##### SUMMARY
When I try to use dict object inside include task - with_dict tells "ERROR! with_dict expects a dict"

##### STEPS TO REPRODUCE
In my task/main.yml I'm including another file for specific task:

```
- name: Calling env.yml for different users
  include: env.yml user={{item}}
  with_items: "{{ users }}"
```
In env.yml code is:
```
- name: Add env
  lineinfile: dest="/home/{{ user }}/{{ docker_compose_base_dir }}/dev.env" line="{{ item.key }}{{ ":" }} {{ item.value }}"
  with_dict: 
    - "{{ nginx }}"
  become: yes
  become_user: "{{ user }}"
```
dictionary is:
```
nginx:
  FE_PORT: 11115
  FE_SOCKET: 11116
  BE_PORT: 1337
  BE_NGINX_PORT: 11117
```

##### EXPECTED RESULTS
Add lines:
  FE_PORT: 11115
  FE_SOCKET: 11116
  BE_PORT: 1337
  BE_NGINX_PORT: 11117
to dev.env file

##### ACTUAL RESULTS
when I try to use with_items it says:

> TASK [users : Add env] ***********************************************************
> fatal: [localhost]: FAILED! => {"failed": true, "msg": "ERROR! 'dict object' has no attribute 'key'"

change it to with_dict:

> TASK [users : Add env] ********************
> fatal: [localhost]: FAILED! => {"failed": true, "msg": "ERROR! with_dict expects a dict"}

Anyway code:
```
- name: debug
  debug: msg="{{ nginx }}"
  tags: [user,devfiles]
```
shows:

> TASK [users : debug] ***********************************************************
> ok: [localhost] => {
>     "msg": {
>         "BE_NGINX_PORT": 11117, 
>         "BE_PORT": 1337, 
>         "FE_PORT": 11115, 
>         "FE_SOCKET": 11116
>     }
> }

**Comment by dimakievua** [comment]
Found bug. with_dict works only if typing as:
`  with_dict: "{{ testusernginx }}"`
next spelling is causing error:
```
  with_dict: 
     - "{{ testusernginx }}"
```

**Comment by nicoonoclaste** [comment]
@dimakievua This isn't an Ansible bug, in my opinion:
using `with_dict "{{ some_dict }}"`, you are correctly passing a dictionary, whereas what you had in the playbook passes to `with_dict` a list whose only element is a dictionary.

**Comment by rockpenguin** [comment]
I banged my head for over an hour before I found this issue :-) I did the same thing:

```yaml
- name: Print sshd values
    debug:
      msg: "Value {{ item.key }} is {{ item.value.keyword }} and {{ item.value.argument }}"
    with_dict:
      - "{{ hardened_keywords }}"
```
and got the same error message: "with_dict expects a dict". I'm on Ansible 2.3.1.0 (Mac).  Once I placed it inline `with_dict: "{{ hardened_keywords }}"` as suggested by @nbraud it worked like a charm.

The [documentation ](http://docs.ansible.com/ansible/latest/playbooks_loops.html#looping-over-hashes) reflects what @dimakievua had in his playbook. So if it's not a bug, then maybe the documentation can be updated to reflect this? Is anyone allowed to do a PR for the documentation, or that a Red Hat thing - I'll be happy to help update if needed.
